src/main/java/org/threadly/db/aurora/AuroraDataSource.java

```diff
@@ -0,0 +1,321 @@
+package org.threadly.db.aurora;
+
+import java.io.PrintWriter;
+import java.io.Serializable;
+import java.sql.Connection;
+import java.sql.SQLException;
+import java.sql.SQLFeatureNotSupportedException;
+import java.util.Properties;
+import java.util.logging.Logger;
+
+import javax.naming.NamingException;
+import javax.naming.Reference;
+import javax.naming.Referenceable;
+import javax.naming.StringRefAddr;
+import javax.sql.DataSource;
+
+public final class AuroraDataSource implements DataSource, Referenceable, Serializable {
+  
+  private static final String AURORA_PREFIX = "jdbc:mysql:aurora:";
+  
+  private static final long serialVersionUID = 1L;
+  
+  /** The driver to create connections with */
+  private static final NonRegisteringDriver driver;
+  
+  static {
+    driver = new NonRegisteringDriver();
+  }
+  
+  /** Log stream */
+  protected transient PrintWriter logWriter = null;
+  
+  /** Database Name */
+  protected String databaseName = null;
+  
+  /** Character Encoding */
+  protected String encoding = null;
+  
+  /** Hostname */
+  protected String hostName = null;
+  
+  /** Password */
+  protected String password = null;
+  
+  /** The profileSQL property */
+  protected String profileSQLString = "false";
+  
+  /** The JDBC URL */
+  protected String url = null;
+  
+  /** User name */
+  protected String user = null;
+  
+  /** Should we construct the URL, or has it been set explicitly */
+  protected boolean explicitUrl = false;
+  
+  /** Port number */
+  protected int port = 3306;
+  
+  /**
+   * Default no-arg constructor for Serialization
+   */
+  public AuroraDataSource() {
+    // Default constructor for Serialization...
+  }
+  
+  @Override
+  public Logger getParentLogger() throws SQLFeatureNotSupportedException {
+    return null;
+  }
+  
+  @Override
+  public boolean isWrapperFor(Class<?> iface) throws SQLException {
+    return false;
+  }
+  
+  @Override
+  public <T> T unwrap(Class<T> iface) throws SQLException {
+    return null;
+  }
+  
+  @Override
+  public Connection getConnection() throws SQLException {
+    return getConnection(this.user, this.password);
+  }
+  
+  /**
+   * Creates a new connection with the given username and passwordsetPropertiesViaRef
+   * 
+   * @param userID
+   *          the user id to connect with
```

Comment:
I would just move these to the `@param` above to remain consistent with the rest of the project.